conceptnet5/vectors/transforms.py

```diff
@@ -91,7 +98,47 @@ def shrink_and_sort(frame, n, k):
     return shrunk
 
 
-def make_replacements(small_frame, big_frame, limit=100000):
+def build_annoy_tree(frame, tree_depth):
+    """
+    Build a tree to hold a frame's vectors for efficient lookup.
+    """
+    index = AnnoyIndex(frame.shape[1], metric='euclidean')
+    index_map = {}
+    for i, item in enumerate(frame.index):
+        index.add_item(i, frame.loc[item])
+        index_map[i] = item
+    index.build(tree_depth)
+    return index, index_map
+
+
+def make_replacements_faster(small_frame, big_frame, tree_depth=1000, verbose=False):
+    """
+    Create a replacements dictionary to map terms only present in a big frame to the closest term
+    in a small_frame. This is a faster than make_replacements(), because it uses a fast
+    implementation of the approximate nearest neighbor algorithm.
+
+    tree_depth=1000 provides a good balance of speed and accuracy.
+    """
+    intersected = big_frame.loc[small_frame.index].dropna()
+    index, index_map = build_annoy_tree(intersected, tree_depth)
+    replacements = {}
+    for term in big_frame.index:
+        if term not in small_frame.index:
+            most_similar = index.get_nns_by_vector(big_frame.loc[term], 1)[0]
+            replacements[term] = index_map[most_similar]
+
+    if verbose:
```

Comment:
Oh, what I meant was that I wanted to see every nth replacement as they're happening, not all at the end.